### PR TITLE
Retrieve hostname for message-id as private var from 'from' address

### PR DESCRIPTION
**Pull request by leanne63**
## Description

Mail.swift:
- Added a private computed property to provide 'hostname' value. The property retrieves the domain field (after the "@" symbol) from the `User` object's `from` field.
- Removed the `hostname`-related parameters from the `Mail` object initialization method.

SMTP.swift:
- Removed the `hostname` argument from the return call to `Mail` in the `makeMail` function.

## Motivation and Context

This change reverts the current broken API, while still fixing the invalid message-id. It also keeps the user from having to determine what to insert as the domain (right-side) portion of the message-id, keeping the message-id in compliance with [RFC 5322](https://tools.ietf.org/html/rfc5322#section-3.6.4). 

This fixes [issue #41: Message-Id improperly formatted](https://github.com/IBM-Swift/Swift-SMTP/issues/41), also unbreaking the earlier API change.

## How Has This Been Tested?

- The change consists of a private computed variable, used only in the `Mail` struct. I searched for calls to initialize the `Mail` object, which resulted in the change to the `SMTP` struct's `makeMail` function. No other code is initializing a `Mail` struct at this point.
- I ran Swift-SMTP code as part of a server-side Swift app, sending multiple emails to multiple recipients. I checked each email for proper message-id format. I also ensured that none of the messages were bounced as a result of a spam filter on the message-id field.

My testing environment is a MacBook Pro, MacOS Sierra (10.12.6), with Swift 3.1 and Swift Package Manager 3.1.0, including Kitura.

## Checklist:

- [x] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [ ] If applicable, I have updated the documentation accordingly.
- [ ] If applicable, I have added tests to cover my changes.

### CODE REVIEW

**Comment by leanne63** [comment]
Looks like the Travis CI environment isn't set up properly? The failures don't related to code... Is there anything else I need to do at this point?

**Comment by quanvo87** [comment]
No, you don't need to do anything. The way it's set up now, it's pulling credentials from the person who opened the PR in order to clone a private repo that stores testing credentials. We're working on an improvement to this process so that your PR can run on Travis properly.

File: Sources/Mail.swift [hunk 1]
```diff
@@ -19,13 +19,21 @@ import Foundation
...
     private let uuid = UUID().uuidString
+	
+	private var hostname: String {
+		let fullEmail = from.email
+		let atIndex = fullEmail.characters.index(of: "@")
+		let hostStart = fullEmail.index(after: atIndex!)
+		let hostnameVal = fullEmail.substring(from: hostStart)
+		
+		return hostnameVal
+	}
```
Comment (quanvo87) [inline Sources/Mail.swift]: Can some simple tests be added for this function?

File: Sources/SMTP.swift [hunk 2]
```diff
@@ -49,7 +49,7 @@ public struct SMTP {
     private let authMethods: [AuthMethod]
     private let domainName: String
     private let accessToken: String?
-    private let timeout: UInt
+    private let timeout: Int
```
Comment (quanvo87) [inline Sources/SMTP.swift]: It looks like this PR was opened against [v1.1.0](https://github.com/IBM-Swift/Swift-SMTP/releases/tag/v1.1.0). Can you rebase against [v1.1.1](https://github.com/IBM-Swift/Swift-SMTP/releases/tag/v1.1.1)? This should fix these discrepancies.

File: Sources/SMTP.swift [hunk 3]
```diff
@@ -90,7 +90,7 @@ public struct SMTP {
                 authMethods: [AuthMethod] = [],
                 domainName: String = "localhost",
                 accessToken: String? = nil,
-                timeout: UInt = 10) {
+                timeout: Int = 10) {
```
Comment (quanvo87) [inline Sources/SMTP.swift]: Same as above.

File: Sources/SMTP.swift [hunk 4]
```diff
@@ -205,11 +167,12 @@ public struct SMTP {
...
-                      timeout: timeout) { (loginResult) in
-                        switch loginResult {
-                        case .failure(let error):
+                      timeout: timeout) { (socket, error) in
+                        if let error = error {
                             completion?([], mails.map { ($0, error) })
-                        case .success(let socket):
+                            return
+                        }
+                        if let socket = socket {
```
Comment (quanvo87) [inline Sources/SMTP.swift]: Same as above.

File: Sources/SMTP.swift [hunk 5]
```diff
@@ -220,4 +183,41 @@ public struct SMTP {
...
+                         additionalHeaders: [String: String] = [:]) -> Mail {
+        return Mail(from: from,
+                    to: to,
+                    cc: cc,
+                    bcc: bcc,
+                    subject: subject,
+                    text: text,
+                    attachments: attachments,
+                    additionalHeaders: additionalHeaders)
+    }
```
Comment (quanvo87) [inline Sources/SMTP.swift]: With your changes to the `Mail` struct, I think this extra `makeMail()` function can be removed.

**Comment by quanvo87** [comment]
Once the PR has been approved, I will have to make a commit on top of it in order for Travis to run correctly.

**Comment by leanne63** [comment]
Working on this, @quanvo87 !

**Comment by leanne63** [comment]
I've rebased the branch to 1.1.1 and removed the makeMail() function, Quan. I've begun working on tests, but am unable to run them at this time. It appears there are other UInt changes that need to be made (in Login.swift and SMTPSocket.swift) to complete the `timeout` changes. Then, there's an error regarding the number of parameters in a closure.

These aren't anything I touched in my changes. Here are some of the messages:

````
Swift-SMTP/Sources/SMTP.swift:170:32: error: cannot convert value of type 'Int' to expected argument type 'UInt'
                      timeout: timeout) { (socket, error) in
                               ^~~~~~~
                               UInt(  )
````

````
Swift-SMTP/Sources/SMTP.swift:162:13: warning: no calls to throwing functions occur within 'try' expression
            try Login(hostname: hostname,
            ^
Swift-SMTP/Sources/SMTP.swift:182:11: warning: 'catch' block is unreachable because no errors are thrown in 'do' block
        } catch {
          ^
````

````
Swift-SMTP/Sources/Login.swift:79:63: error: cannot convert value of type 'Int' to expected argument type 'UInt'
        try socket.connect(to: hostname, port: port, timeout: timeout)
                                                              ^~~~~~~
                                                              UInt(  )
````

````
Swift-SMTP/Sources/SMTP.swift:170:43: error: contextual closure type '(LoginResult) -> Void' expects 1 argument, but 2 were used in closure body
                      timeout: timeout) { (socket, error) in
                                          ^
````

**Comment by quanvo87** [comment]
I'm not sure what's going on haha. I opened [PR #46](https://github.com/IBM-Swift/Swift-SMTP/pull/46) against a fresh branch of v1.1.1, adding your changes, and everything seems to be working fine. Can we work from that [branch](https://github.com/IBM-Swift/Swift-SMTP/tree/fix_message_id) from here on? I will make sure due credit for your contributions is given.

And for the tests, a few simple happy/failure paths is fine.

**Pull request merged**